worlds/sc2/rules.py

```diff
@@ -1464,13 +1707,15 @@ def night_terrors_requirement(self, state: CollectionState) -> bool:
                     )
                 )
             )
+            and self.terran_army_weapon_armor_upgrade_count(state) >= 2
         )
 
     def flashpoint_far_requirement(self, state: CollectionState) -> bool:
         return (
             self.terran_competent_comp(state)
             and self.terran_mobile_detector(state)
             and self.terran_defense_rating(state, True, False) >= 6
+            and self.terran_army_weapon_armor_upgrade_count(state) >= 2
```

Comment:
I haven't really had much issue with flashpoint, I'd say make this just +1, and in competent comp only request +2 for infantry.